catalog/plugin/catalog-plugin-auditclassifiedaccess/src/main/java/org/codice/alliance/catalog/plugin/auditclassified/AuditClassifiedAccessPlugin.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.catalog.plugin.auditclassified;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.common.audit.SecurityLogger;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AuditClassifiedAccessPlugin implements PostQueryPlugin {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AuditClassifiedAccessPlugin.class);
+
+  private Map<String, List<String>> classifiedValuesMap = new HashMap<String, List<String>>();
+
+  public QueryResponse process(QueryResponse input)
+      throws PluginExecutionException, StopProcessingException {
+
+    List<Result> results = input.getResults();
+
+    results.stream().map(Result::getMetacard).filter(Objects::nonNull).forEach(this::handleAudits);
+
+    LOGGER.trace("Response went through the Audit Classified Access Plugin");
+    return input;
+  }
+
+  private void handleAudits(Metacard metacard) {
+    boolean isClassified = false;
+    Iterator<Entry<String, List<String>>> it = classifiedValuesMap.entrySet().iterator();
+
+    while (it.hasNext() && !isClassified) {
+      Entry<String, List<String>> entry = it.next();
+      if (hasClassifiedValues(entry.getValue(), entry.getKey(), metacard)) {
+        isClassified = true;
+      }
+    }

```

Comment:
As far as I can tell, this cannot be implemented with streams and still terminate the loop early on the first match.